public/app/features/variables/pickers/OptionsPicker/OptionsPicker.tsx

```diff
@@ -16,125 +16,125 @@ import { formatVariableLabel } from '../../shared/formatVariable';
+    toggleOption,
+    toggleAndFetchTag,
+    navigateOptions,
+  };
 
-interface ConnectedProps {
-  picker: OptionsPickerState;
-}
+  const mapStateToProps = (state: StoreState) => ({
+    picker: state.templating.optionsPicker,
+  });
 
-interface DispatchProps {
-  showOptions: typeof showOptions;
-  commitChangesToVariable: typeof commitChangesToVariable;
-  toggleAllOptions: typeof toggleAllOptions;
-  toggleOption: typeof toggleOption;
-  toggleAndFetchTag: typeof toggleAndFetchTag;
-  filterOrSearchOptions: typeof filterOrSearchOptions;
-  navigateOptions: typeof navigateOptions;
-}
+  const connector = connect(mapStateToProps, mapDispatchToProps);
 
-type Props = OwnProps & ConnectedProps & DispatchProps;
+  interface OwnProps extends VariablePickerProps<Model> {}
 
-export class OptionsPickerUnconnected extends PureComponent<Props> {
-  onShowOptions = () => this.props.showOptions(this.props.variable);
-  onHideOptions = () => this.props.commitChangesToVariable(this.props.onVariableChange);
+  type Props = OwnProps & ConnectedProps<typeof connector>;
 
-  onToggleOption = (option: VariableOption, clearOthers: boolean) => {
-    const toggleFunc = this.props.variable.multi ? this.onToggleMultiValueVariable : this.onToggleSingleValueVariable;
-    toggleFunc(option, clearOthers);
-  };
+  class OptionsPickerUnconnected extends PureComponent<Props> {
+    onShowOptions = () => this.props.showOptions(this.props.variable);
+    onHideOptions = () => this.props.commitChangesToVariable(this.props.onVariableChange);
 
-  onToggleSingleValueVariable = (option: VariableOption, clearOthers: boolean) => {
-    this.props.toggleOption({ option, clearOthers, forceSelect: false });
-    this.onHideOptions();
-  };
+    onToggleOption = (option: VariableOption, clearOthers: boolean) => {
+      const toggleFunc = this.hasMultiPropertyAssigned(this.props.variable)
+        ? this.onToggleMultiValueVariable
+        : this.onToggleSingleValueVariable;
+      toggleFunc(option, clearOthers);
+    };
 
-  onToggleMultiValueVariable = (option: VariableOption, clearOthers: boolean) => {
-    this.props.toggleOption({ option, clearOthers, forceSelect: false });
-  };
+    onToggleSingleValueVariable = (option: VariableOption, clearOthers: boolean) => {
+      this.props.toggleOption({ option, clearOthers, forceSelect: false });
+      this.onHideOptions();
+    };
 
-  render() {
-    const { variable, picker } = this.props;
-    const showOptions = picker.id === variable.id;
+    onToggleMultiValueVariable = (option: VariableOption, clearOthers: boolean) => {
+      this.props.toggleOption({ option, clearOthers, forceSelect: false });
+    };
 
-    return (
-      <div className="variable-link-wrapper">
-        {this.renderLink(showOptions, variable)}
-        {this.renderOptions(showOptions, picker)}
-      </div>
-    );
-  }
+    hasMultiPropertyAssigned = (variable: any) => {
+      return variable.hasOwnProperty('multi') && variable.multi === true;
```

Comment:
I was wondering if we could move the `multi` property to `VariableWithOptions`, could save an extra check for the `multi` prop, plus in the current types `multi` is only either true or undefined. Or maybe even combine `VariableWithOptions` and `VariableWithMultiSupport` into one type?

Alternatively, can we use `isMulti` guard here instead?

Reply:
I see your point, I'll see if I can use `isMulti`. Good catch!

Reply:
I think the change with the merging of types is a subject for another PR.

Reply:
Fixed!